### DISCUSSION TITLE
11. IoC와 DI에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by KIMSEI1124** [Study-Diary/Computer-Science#48] [Spring]
- 후보 없이 특정 기능을 하는 클래스가 딱 한 개하면, 구체 클래스를 그냥 사용해도 되지 않나요? 그럼에도 불구하고 왜 Spring에선 Bean을 사용 할까요?
- Spring의 Bean 생성 주기에 대해 설명해 주세요.
- 프로토타입 빈은 무엇인가요?

**Comment by jiwon83** [comment]
Setter 주입을 사용할 상황은 언제일까요?

**Comment by sjhjack** [comment]
인터페이스에 빈으로 등록된 구현체가 여러개 있을 경우, 어떤 구현체가 주입될까요?
